src/RadixDlt.NetworkGateway.DataAggregator/Workers/NodeWorkers/NodeTransactionFetchWorker.cs

```diff
@@ -236,10 +238,31 @@ private async Task<FetchedTransactions> FetchTransactions(long fromStateVersion,
             fromStateVersion
         );
 
-        var (apiResponse, fetchTransactionsMs) = await CodeStopwatch.TimeInMs(async () => await _transactionStreamReaderFactory().GetTransactionStream(fromStateVersion, count, token));
-        var transactions = apiResponse.Data.Transactions;
+        var (apiResponse, fetchTransactionsMs) = await CodeStopwatch.TimeInMs(async () =>
+        {
+            return await CoreApiErrorWrapper.ResultOrError<CoreClient.ApiResponse<CoreModel.StreamTransactionsResponse>, CoreModel.StreamTransactionsErrorResponse>(() =>
+            {
+                return _transactionStreamReaderFactory().GetTransactionStream(fromStateVersion, count, token);
+            });
+        });
 
-        await _observers.ForEachAsync(x => x.TransactionsFetched(NodeName, transactions, fetchTransactionsMs));
+        if (apiResponse.Failed)
+        {
+            var ex = apiResponse.FailureResponse.OriginalApiException;
+
+            if (apiResponse.FailureResponse.Details is CoreModel.RequestedStateVersionOutOfBoundsErrorDetails oob)
+            {
+                _logger.LogWarning(ex, "Requested state version past the maximal ledger state version of {MaxLedgerStateVersion}", oob.MaxLedgerStateVersion);
+            }
+            else
+            {
+                _logger.LogError(ex, "Unknown response error: {Message}", apiResponse.FailureResponse.Message);
```

Comment:
IMO we should just throw here instead of logging. The Worker's error logging and back-off logic will then kick in, which is what we want if we're hammering a Core API which is struggling.

Reply:
Agree.